services/__tests__/static.test.js

```diff
@@ -416,4 +416,163 @@ describe('getStaticImage', () => {
+        coordinates: [12, 13],
+        zoom: 3
+      },
+      addlayer: {
+        id: 'tall-buildings',
+        type: 'fill',
+        source: 'composite',
+        'source-layer': 'building',
+        filter: [
+          'all',
+          ['>=', ['get', 'height'], 150],
+          ['match', ['get', 'underground'], ['false'], true, false]
+        ],
+        paint: { 'fill-color': '%235E8DFF', 'fill-opacity': 0.5 }
+      },
+      before_layer: 'tunnel-street-minor-low'
+    });
+    expect(tu.requestConfig(service)).toEqual({
+      method: 'GET',
+      path: '/styles/v1/:ownerId/:styleId/static/12,13,3/200x300',
+      query: {
+        addlayer: {
+          id: 'tall-buildings',
+          type: 'fill',
+          source: 'composite',
+          'source-layer': 'building',
+          filter: [
+            'all',
+            ['>=', ['get', 'height'], 150],
+            ['match', ['get', 'underground'], ['false'], true, false]
+          ],
+          paint: { 'fill-color': '%235E8DFF', 'fill-opacity': 0.5 }
+        },
+        before_layer: 'tunnel-street-minor-low'
+      },
+      params: { ownerId: 'mapbox', styleId: 'streets-v10' }
+    });
+  });
+
+  test('Auto extent cannot be used with style parameters and no overlay', () => {
+    expect(() => {
+      service.getStaticImage({
+        ownerId: 'mapbox',
+        styleId: 'streets-v10',
+        width: 200,
+        height: 300,
+        position: 'auto',
+        addlayer: {
+          id: 'tall-buildings',
+          type: 'fill',
+          source: 'composite',
+          'source-layer': 'building',
+          filter: [
+            'all',
+            ['>=', ['get', 'height'], 150],
+            ['match', ['get', 'underground'], ['false'], true, false]
+          ],
+          paint: { 'fill-color': '%235E8DFF', 'fill-opacity': 0.5 }
+        },
+        before_layer: 'tunnel-street-minor-low'
+      });
+    }).toThrow(
+      /Auto extent cannot be used with style parameters and no overlay/
+    );
+  });
+
+  test('addlayer and setfilter cannot be used together unless also using an overlay', () => {
+    expect(() => {
+      service.getStaticImage({
+        ownerId: 'mapbox',
+        styleId: 'streets-v10',
+        width: 200,
+        height: 300,
+        position: {
+          coordinates: [12, 13],
+          zoom: 3
+        },
+        addlayer: {
+          id: 'tall-buildings',
+          type: 'fill',
+          source: 'composite',
+          'source-layer': 'building',
+          filter: [
+            'all',
+            ['>=', ['get', 'height'], 150],
+            ['match', ['get', 'underground'], ['false'], true, false]
+          ],
+          paint: { 'fill-color': '%235E8DFF', 'fill-opacity': 0.5 }
+        },
+        before_layer: 'tunnel-street-minor-low',
+        setfilter: ['in', 'code', 'CA'],
+        layer_id: 'tunnel-street-minor-low'
+      });
+    }).toThrow(/addlayer and setfilter cannot be used in the same request/);
+  });
```

Comment:
The test itself is correct, but the description is not 😬 

You can't use `setfilter` and `addlayer` together _even if_ you're using an overlay.

Reply:
🤦‍♂ thanks for catching that!